mathesar/tests/views/api/test_table_api.py

```diff
@@ -873,3 +885,75 @@ def test_table_viewset_checks_cache(client):
     with patch.object(reflection, 'reflect_tables_from_schema') as mock_reflect:
         client.get('/api/v0/tables/')
     mock_reflect.assert_called()
+
+
+def test_table_retrieve_no_primary_key(create_table, client):
+    table_name = 'NASA Table Retrieve'
+    new_table_name = 'NASA Table Retrieve New'
+    table = create_table(table_name)
+
+    body = {'name': new_table_name, 'num_primary_keys': 0}
+    try:
+        client.get(f'/api/v0/tables/{table.id}/', body)
+    except NotUniquePrimaryKey as e:
+        assert e
+
+
+def test_table_delete_no_primary_key(create_table, client):
+    table_name = 'NASA Table Delete'
+    new_table_name = 'NASA Table Delete New'
+    table = create_table(table_name)
+
+    body = {'name': new_table_name, 'num_primary_keys': 0}
+    try:
+        client.delete(f'/api/v0/tables/{table.id}/', body)
+    except NotUniquePrimaryKey as e:
+        assert e
+
+
+def test_table_partial_update_no_primary_key(create_table, client):
+    table_name = 'NASA Table Partial Update'
+    new_table_name = 'NASA Table Partial Update New'
+    table = create_table(table_name)
+
+    body = {'name': new_table_name, 'num_primary_keys': 0}
+    try:
+        client.patch(f'/api/v0/tables/{table.id}/', body)
+    except NotUniquePrimaryKey as e:
+        assert e
+
+
+def test_table_retrieve_multiple_primary_keys(create_table, client):
+    table_name = 'NASA Table Retrieve'
+    new_table_name = 'NASA Table Retrieve New'
+    table = create_table(table_name)
```

Comment:
You cannot use the existing `create_table` function here, since it creates a table with a single primary key. You'll need to create a table with a composite primary key manually. Also applies to other `multiple_primary_key` functions below.